### PR TITLE
[ISSUE-321][Feature]  Worker trigger shutdown hook when run command stop-worker.sh

### PR DESCRIPTION
**Pull request by AngersZhuuuu**
# [BUG]/[FEATURE] title

### What changes were proposed in this pull request?
Add an shutdown hook in Worker's shutdown progress.

Logs after add some stopping log
```
22/08/12 10:30:05,240 INFO [shutdown-hook-0] Worker: Stopping RSS Worker.
22/08/12 10:30:05,240 INFO [shutdown-hook-0] Worker: Stopping forwardMessageScheduler
22/08/12 10:30:05,241 INFO [shutdown-hook-0] Worker: Stopping replicateThreadPool
22/08/12 10:30:05,242 INFO [shutdown-hook-0] Worker: Stopping commitThreadPool
22/08/12 10:30:05,243 INFO [shutdown-hook-0] Worker: Stopping asyncReplyPool
22/08/12 10:30:05,244 INFO [shutdown-hook-0] Worker: Stopping partitionsSorter
22/08/12 10:30:05,244 INFO [shutdown-hook-0] PartitionFilesSorter: Start close PartitionFilesSorter
22/08/12 10:30:05,244 WARN [Thread-1] PartitionFilesSorter: Sort thread is shutting down, detail :
java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.reportInterruptAfterWait(AbstractQueuedSynchronizer.java:2014)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2048)
	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
	at com.aliyun.emr.rss.service.deploy.worker.PartitionFilesSorter.lambda$new$1(PartitionFilesSorter.java:77)
	at java.lang.Thread.run(Thread.java:748)
22/08/12 10:30:05,244 INFO [shutdown-hook-0] Worker: Stopping partitionLocationInfo
22/08/12 10:30:05,246 INFO [shutdown-hook-0] PartitionLocationInfo: Start close PartitionLocationInfo
22/08/12 10:30:05,247 INFO [shutdown-hook-0] Worker: Stopping localStorageManager
22/08/12 10:30:05,253 INFO [shutdown-hook-0] Worker: Stopping rssHARetryClient
22/08/12 10:30:05,255 INFO [shutdown-hook-0] Worker: Stopping replicateServer
22/08/12 10:30:05,259 INFO [shutdown-hook-0] Worker: Stopping fetchServer
22/08/12 10:30:05,264 INFO [shutdown-hook-0] Worker: RSS Worker is stopped.
```

### Related issues.
https://github.com/alibaba/RemoteShuffleService/issues/321

### How was this patch tested?

/cc @related-reviewer

/assign @main-reviewer

### CODE REVIEW

File: common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+
+  static class HookEntry {
+    private final Runnable hook;
+    private final int priority;
+    private final long timeout;
+    private final TimeUnit unit;
+
+    HookEntry(Runnable hook, int priority) {
+      this(hook, priority,
+          getShutdownTimeout(new RssConf()),
```
Comment (waitinfuture) [inline common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java:171]: Same as above.

File: common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+   * Shutdown the executor thread itself.
+   * @param conf the configuration containing the shutdown timeout setting.
+   */
+  private static void shutdownExecutor(final RssConf conf) {
+    try {
+      EXECUTOR.shutdown();
+      long shutdownTimeout = getShutdownTimeout(conf);
+      if (!EXECUTOR.awaitTermination(
+          shutdownTimeout,
+          TIME_UNIT_DEFAULT)) {
```
Comment (waitinfuture) [inline common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java:135]: getShutdownTimeout returns time in milliseconds, but here use default of seconds, should make them consistent.

File: common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java (hunk 1 above)
Comment (AngersZhuuuu) [reply common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java:171]: Done

File: common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java (hunk 2 above)
Comment (AngersZhuuuu) [reply common/src/main/java/com/aliyun/emr/rss/common/utils/ShutdownHookManager.java:135]: Done

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**